src/transformers/trainer.py

```diff
@@ -346,6 +346,9 @@ def __init__(
         self._memory_tracker = TrainerMemoryTracker(self.args.skip_memory_metrics)
         self._memory_tracker.start()
 
+        # Fillers for storing the original eval_ratio for when we use the batch size finder
+        self._step_ratios = {}
```

Comment:
This shouldn't exist anymore since we keep the values as they were.

Reply:
But it doesn't because the state is initialized here https://github.com/huggingface/transformers/blob/main/src/transformers/trainer.py#L1627 (and thus completely reset), so we need to have this reference outside of it otherwise it will be lost.

Reply:
What will be lost? We shouldn't change any of the training arguments so nothing can be lost.

Reply:
Ah yes sorry that's right